card/lib/card/query.rb

```diff
@@ -163,14 +196,22 @@ def interpret_by_key clause
     end
```

Comment:
wouldn't that just alter the local variable definition of val without changing the hash?